### PR TITLE
refactor layout station

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are refactoring `LayoutStation` in MassaStation.

### CODE REVIEW

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    activePage,
+    customClass,
+  } = props;
+
+  const [storedTheme] = useLocalStorage<string>(
+    'massa-station-theme',
+    'theme-dark',
+  );
+  const [selectedTheme, setSelectedTheme] = useState(
+    overwrittenTheme || storedTheme || 'theme-dark',
```
Comment (Pierregvx) [inline web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:41]: when do we overwrite the theme ? by curiosity

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+} from '@massalabs/react-ui-kit';
+
+export interface LayoutStationProps {
+  children?: ReactNode;
+  navigator?: Navigator;
+  logo?: ReactNode;
+  onSetTheme?: () => void;
+  activePage: string;
+  customClass?: string;
+  overwrittenTheme?: string;
```
Comment (mazmassa) [inline web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:23]: ...do we need that!? 🤔

File: web/massastation/src/pages/Search/Search.tsx [hunk 3]
```diff
@@ -36,6 +42,27 @@ export function Search() {
     }
   }, [error, navigate]);
 
+  // const { data: accounts = [] } = useResource<AccountObject[]>(
```
Comment (mazmassa) [inline web/massastation/src/pages/Search/Search.tsx:45]: All these comments will be removed!? 🤔

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx (hunk 1 above)
Comment (maxime-carabina) [reply web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:41]: This was already present on the ui-kit. I think it's just to override theme if one page is different by other.

File: web/massastation/src/pages/Search/Search.tsx [hunk 4]
```diff
@@ -36,6 +42,27 @@ export function Search() {
...
+
+  // // const accountsItems = accounts.map((account) => ({
+  //   // icon: <Identicon username={account.nickname} size={32} />,
+  //   // item: account.nickname,
+  //   // onClick: () => setNickname(account.nickname),
+  // }));
+
+  // const selectedAccountKey: number = parseInt(
+  //   Object.keys(accounts).find(
+  //     (_, idx) => accounts[idx].nickname === nickname,
```
Comment (pivilartisant) [inline web/massastation/src/pages/Search/Search.tsx:61]: big comment is legacy ?

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx (hunk 2 above)
Comment (maxime-carabina) [reply web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:23]: I need activePage, but other you're right

File: web/massastation/src/pages/Search/Search.tsx (hunk 3 above)
Comment (maxime-carabina) [reply web/massastation/src/pages/Search/Search.tsx:45]: Oh... sorry for that

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+  const [storedTheme] = useLocalStorage<string>(
+    'massa-station-theme',
+    'theme-dark',
+  );
+  const [selectedTheme, setSelectedTheme] = useState(
+    overwrittenTheme || storedTheme || 'theme-dark',
+  );
+
+  const searchIsActive = activePage === 'search';
```
Comment (pivilartisant) [inline web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:44]: `const searchIsActive = activePage === PAGES.SEARCH;`

File: web/massastation/src/pages/Search/Search.tsx [hunk 6]
```diff
@@ -8,6 +8,12 @@ import Upload from '../../components/SidePanels/Upload/Upload';
 import DomainModelItem from './DomainModelItem/DomainModelItem';
 import { routeFor } from '../../utils';
 
+// import { AccountObject } from '../../models/AccountModel';
+// import { useAccountStore } from '../../store/store';
+// import { URL } from '../../const/url/url';
```
Comment (pivilartisant) [inline web/massastation/src/pages/Search/Search.tsx:13]: legacy comments ?

File: web/massastation/src/pages/Search/Search.tsx [hunk 7]
```diff
@@ -8,6 +8,12 @@ import Upload from '../../components/SidePanels/Upload/Upload';
...
+
+// import { Dropdown, Identicon } from '@massalabs/react-ui-kit';
+
 const FAVORITES = ['flappy', 'psychedelic', 'flappynathana1'];
```
Comment (pivilartisant) [inline web/massastation/src/pages/Search/Search.tsx:17]: add these values in a unique source of truth like massaConstants.ts

Comment (maxime-carabina) [reply web/massastation/src/pages/Search/Search.tsx:17]: > add these values in a unique source of truth like massaConstants.ts

In which folder can I create this file ?

File: web/massastation/src/pages/Search/Search.tsx (hunk 6 above)
Comment (maxime-carabina) [reply web/massastation/src/pages/Search/Search.tsx:13]: yes sorry

File: web/massastation/src/pages/Search/Search.tsx (hunk 7 above)
Comment (pivilartisant) [reply web/massastation/src/pages/Search/Search.tsx:17]: it is already created is src/utils

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx (hunk 5 above)
Comment (maxime-carabina) [reply web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:44]: thanks for that

File: web/massastation/src/pages/Search/Search.tsx (hunk 4 above)
Comment (maxime-carabina) [reply web/massastation/src/pages/Search/Search.tsx]: Done

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  Dropdown,
+  Identicon,
+} from '@massalabs/react-ui-kit';
+
+export interface LayoutStationProps {
+  children?: ReactNode;
+  navigator?: Navigator;
+  logo?: ReactNode;
+  onSetTheme?: () => void;
+  storedTheme?: string;
```
Comment (mazmassa) [inline web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:20]: Can we change that just for theme!?

File: web/massastation/src/utils/massaConstants.ts [hunk 9]
```diff
@@ -7,3 +7,6 @@ export const PLUGIN_STOP = 'stop';
 export const PLUGIN_UPDATE = 'update';
 
 export const isZipFile = /^(http)[^\s]*\.zip$/i;
+
+// List of websites that are allowed to be added to favorites
+export const FAVORITES_WEBSITES = ['flappy', 'psychedelic', 'flappynathana1'];
```
Comment (mazmassa) [inline web/massastation/src/utils/massaConstants.ts:12]: Let's move that to an specific folder for const. I would say the same we have in the wallet. Like that we can have a kind of standard.

web-frontend/src/const

Comment (mazmassa) [reply web/massastation/src/utils/massaConstants.ts:12]: Feel free to close here when it's done!

**Review by mazmassa — APPROVED** [review]
LGTM Max! Thanks for that! Well done!

👏

File: web/massastation/src/utils/massaConstants.ts (hunk 9 above)
Comment (pivilartisant) [reply web/massastation/src/utils/massaConstants.ts:12]: By bad i told @maxime-carabina tu put it here !! :/

**Review by pivilartisant — APPROVED** [review]

File: web/massastation/src/utils/massaConstants.ts (hunk 9 above)
Comment (maxime-carabina) [reply web/massastation/src/utils/massaConstants.ts:12]: > Let's move that to an specific folder for const. I would say the same we have in the wallet. Like that we can have a kind of standard.
> 
> web-frontend/src/const

Already present in `src/utils/` 😃

File: web/massastation/src/layouts/LayoutStation/LayoutStation.tsx [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+    setSelectedTheme(theme);
+
+    onSetTheme?.(theme);
+  }
+
+  const { data: accounts = [] } = useResource<AccountObject[]>(
+    `${URL.WALLET_BASE_API}/${URL.WALLET_ACCOUNTS}`,
+  );
+
+  // Store
```
Comment (Thykof) [inline web/massastation/src/layouts/LayoutStation/LayoutStation.tsx:42]:
```suggestion
```

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Pull request merged**